frontend/src/app/commonComponents/TextInput.jsx

```diff
@@ -3,46 +3,33 @@ import PropTypes from "prop-types";
 import classnames from "classnames";
 import useUniqueId from "../commonComponents/useUniqueIds";
 
-const TextInput = ({
-  value = "",
-  label,
-  name,
-  placeholder,
-  onChange,
-  type,
-  addClass,
-}) => {
-  let [newId] = useUniqueId("textinput", 1);
-
+const TextInput = ({ label, type = "text", ...inputProps }) => {
+  const [newId] = useUniqueId("textinput", 1);
+  const id = inputProps.id || newId;
+  const props = { id, type, ...inputProps };
+  // merge in css classes
+  props.className = classnames("usa-input", inputProps.className);
   const labelElem = label ? (
-    <label className="usa-label" htmlFor={newId}>
+    <label className="usa-label" htmlFor={id}>
       {label}
     </label>
   ) : null;
 
   return (
-    <div className={classnames("prime-text-input", addClass)}>
+    <React.Fragment>
```

Comment:
This is losing the `prime-text-input` class.

Reply:
Yes, this was causing layout issues in Settings and other places. 

Here's the css class:
```
.prime-text-input,
.prime-dropdown {
  margin-left: 2em; }
```

I think the better approach is to have the calling code add the classname to the element itself via the params or put it's own enclosing div if it needs it.

Also, using block level elements like `<div>` or `<p>` can be problematic for layouts using the component. I think this is partly caused by our inconsistent usage within the project while we prototype things.

Reply:
Yeah, the lack of app consistency is really biting us here. we have several "forms" that are taking very different approaches to design.